File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
#include <vector>

class Solution {
public:
    void merge(std::vector<int>& nums1, int m, std::vector<int>& nums2, int n) {
        std::vector<int> tmp(nums1.begin(), nums1.begin() + m); // Copy the first m elements of nums1 to tmp
        int i = 0; // Index for tmp
        int j = 0; // Index for nums2
        int k = 0; // Index for nums1

        while (i < m && j < n) {
            if (tmp[i] <= nums2[j]) {
                nums1[k] = tmp[i];
                i++;
            } else {
                nums1[k] = nums2[j];
                j++;
            }
            k++;
        }

        // Copy the remaining elements from tmp or nums2 if any
        while (i < m) {
            nums1[k] = tmp[i];
            i++;
            k++;
        }

        while (j < n) {
            nums1[k] = nums2[j];
            j++;
            k++;
        }
    }
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.go
func merge(nums1 []int, m int, nums2 []int, n int)  {
    i := m - 1 // index for nums1 starts at m (end)
    j := n - 1 // index for nums2 starts at n (end)
    k := len(nums1) - 1 // index for the empty spaces in nums1 (end)
    
    for {
        if i < 0 || j < 0 {
            break
        }
        
        if nums1[i] <= nums2[j] {
            nums1[k] = nums2[j]
            j--
        }else {
            nums1[k] = nums1[i]
            i--
        }
        k--
    }
    
    // Copy any remaining elements from nums2 to nums1
    for j >= 0 {
        nums1[k] = nums2[j]
        j--
        k--
    }
}